routes/routes.go

```diff
@@ -0,0 +1,195 @@
+package routes
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+type pokemon struct {
+	Id    int    `json:"id"`
+	Name  string `json:"name"`
+	Power int    `json:"power"`
+}
+
+type apiResponse struct {
+	Count    int      `json:"count"`
+	Next     string   `json:"next"`
+	Previous []string `json:"previous"`
+	Results  []struct {
+		Name string `json:"name"`
+		URL  string `json:"url"`
+	} `json:"results"`
+}
+
+func readFile(fileName string) [][]string {
+
+	f, err := os.Open(fileName)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer f.Close()
+
+	csvReader := csv.NewReader(f)
+	data, err := csvReader.ReadAll()
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return data
+
+}
+
+func readGeneric(fileName string) map[string]map[string]string {
+
+	f, _ := os.Open(fileName)
+
+	defer f.Close()
+
+	csvReader := csv.NewReader(f)
+	data, _ := csvReader.ReadAll()
+
+	elements := make(map[string]map[string]string)
+	headers := data[0]
+
+	for _, element := range data[1:] {
+		elements[element[0]] = map[string]string{}
+		for i, header := range headers {
+			elements[element[0]][header] = element[i]
+		}
+	}
+
+	return elements
+
+}
+
+func makeFile(headers []string, data apiResponse) {
+
+	f, err := os.Create("returnedData.csv")
+
+	if err != nil {
+		log.Fatalf("failed creating file: %v", err)
+	}
+
+	defer f.Close()
```

Comment:
you can define anonymous functions to catch the `f.Close()` returning error if something went wrong at this step, like this example:

```
defer func(){
  err := f.Close()
  if err != nil {
    log.Println(err)
  }
}()
```